### PR TITLE
"UseOfBasicAuth" Finding

### PR DESCRIPTION
**Pull request by Russ-Corsha**
Before you submit your PR, make sure you check each of the following:

1. [x] I have read the [Contributing Guidelines](./CONTRIBUTING.md) and signed the [Contributor License Agreement](./CLA.md).
2. [x] My code is clean and ready-to-PR, including being free of lint errors and dead code
3. [x] I have **tested** my change / code and am confident that it works
4. [x] I have filled in the template below completely and accurately

##### Changelog
Add in the "UseOfBasicAuth" Finding.  Added jest test to check the logic and added sql traffic to test the psql query. 

##### Testing
Used skaffold to set up the database, added traffic with basic auth, and ran my call against the db.

##### Unit Tests
Jest unit test: ui/src/__tests__/analysis/useOfBasicAuth.test.ts

### CODE REVIEW

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: ui/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
 		"check-stylelint-prettier-conflicts": "stylelint-config-prettier-scss-check",
 		"storybook": "start-storybook -p 6006",
 		"build-storybook": "build-storybook",
-		"test": "jest --watch",
+		"test": "jest --watchAll",
```
Comment (Russ-Corsha) [inline ui/package.json:13]: watch wont trigger on tests that havent changed

File: scripts/cast-ui-local-dev-server.sh [hunk 2]
```diff
@@ -48,11 +48,11 @@ then
 fi
 
 cat > ./ui/.env.local <<HERE
-PGSQL_USER=postgres
-PGSQL_PASSWORD=password
-PGSQL_HOST=127.0.0.1
-PGSQL_PORT=5432
-PGSQL_DATABASE=traffic
+PGUSER=postgres
```
[the remark this answers is not in the document]
Comment (Russ-Corsha) [reply scripts/cast-ui-local-dev-server.sh]: yes, done

File: ui/src/__tests__/analysis/useOfBasicAuth.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/* Copyright 2022 Corsha.
```
Comment (DavidCorsha) [inline ui/src/__tests__/analysis/useOfBasicAuth.test.ts:1]:
```suggestion
/* Copyright 2023 Corsha.
```

File: ui/src/lib/findings.ts [hunk 4]
```diff
@@ -78,10 +78,12 @@ export type IFinding<Type extends string, DataType extends Record<string, any>>
   data: DataType;
 };
 
+export type Proto = "tcp" | "udp" | "unknown";
```
Comment (DavidCorsha) [inline ui/src/lib/findings.ts:81]: Is this used elsewhere

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: ui/src/lib/findings.ts (hunk 4 above)
Comment (Russ-Corsha) [reply ui/src/lib/findings.ts:81]: no, though it may be nice to add a `toProto(string)` method in the future.  removed

**Review by DavidCorsha — CHANGES_REQUESTED** [review]

File: ui/src/lib/analysis/useOfBasicAuth.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            URI: row.absolute_uri,
+        }
+      }
+    };
+    return finding;
+  });
+
+  return {
+    id: "reused-auth",
+    title: "Reused Authentication",
```
[the remark this answers is not in the document]
Comment (Russ-Corsha) [reply ui/src/lib/analysis/useOfBasicAuth.ts]: Great catch, changed

**Review by DavidCorsha — APPROVED** [review]

**Pull request merged**